src/client/src/App.js

```diff
@@ -0,0 +1,26 @@
+import { useState, useCallback } from "react";
+import { ListControls } from "./components/ListControls/ListControls";
+import { ListItem } from "./components/ListItem/ListItem";
+import { useItem } from "./hooks/useItem";
+import "./App.css";
+
+function App() {
+  const { getItems } = useItem();
+  const [itemsList, setItemsList] = useState([]);
+
+  const renderItems = useCallback(async () => {
+    setItemsList(await getItems());
+  }, []);
```

Comment:
did you consider having the  useEffect for this be dependent on itemsList so that when these change it renders the items, this way you don't have to pass this in and call it each time.

In general it is not really something that is done anymore managing the state in the server like you are doing here - you should have the items list in a state in the client and not each time you have some operation then make one call to the server to do the operation and then another to fetch the list of new items. 

You can have the list in state in the client and then the modern approach for fast performance is usually waiting for success or failure from the api call and updating the state (in the client) or not accordingly.

You could also have certain things call to get the updated list from the server or refresh it periodically until you have something that pushes updates from teh server to the client , but that's already a more advanced step for the future when you can have 2 differnt users updating the same thing for example.